pipeline/reload.sh

```diff
@@ -2,24 +2,29 @@
 
 COMMAND="${*@Q}"
 
+EXE=_build/default/bin/main.exe
+
 function run {
   cd /app
   sh -c "$COMMAND" &
+  cd /mnt/project
 }
 
 run
 
+dune build --watch bin/main.exe &
+
 while :
 do
-  cd /mnt/project
-
-  inotifywait -q -q -e CLOSE_WRITE \
-    lib/dune lib/*.ml lib/*.mli \
-    bin/dune bin/*.ml bin/*.mli
-
-  dune build bin/main.exe \
+  # NOTE:
+  # 1. inotifywait exits with a failure status code when main.exe gets deleted
+  #    before a rebuild. We want to exit early, when that happens.
+  # 2. Also, we sleep for a short while, to ensure the process has been killed
+  #    before copying the new executable and trying to run it.
+  test -f "${EXE}" \
+  && inotifywait -q -q -e CLOSE_WRITE "${EXE}" \
```

Comment:
I'm a tiny bit worried about what happens when the executable is removed / not there. For example if the first `dune build` fails, then the while loop will test a bit too fast for the executable existence. Can we perhaps use inotifywait to wait for the executable to exists? or add a sleep somewhere?

It looks great otherwise, thank you!